src/internal/pfsload/validator.go

```diff
@@ -66,12 +72,21 @@ func (v *Validator) Validate(client Client, commit *pfs.Commit) (retErr error) {
 		if ci.Commit.Branch.Repo.Type != pfs.UserRepoType || ci.Origin.Kind == pfs.OriginKind_ALIAS {
 			return nil
 		}
-		return validate(client, ci.Commit, v.hash)
+		hash, err := computeCommitHash(client, ci.Commit)
+		if err != nil {
+			return err
+		}
+		if !bytes.Equal(hash, v.hash) {
```

Comment:
Any reason why this was moved out of computeCommitHash? Do you want potentially reuse computeCommitHash elsewhere?

Reply:
yes, I think it will be more useful for later work and it makes more sense to me to separate the compute and check functionality.